control/control.go

```diff
@@ -218,7 +218,22 @@ func New(cfg *Config) *pluginControl {
 	}).Debug("metric catalog created")
 
 	// Plugin Manager
-	c.pluginManager = newPluginManager(OptSetPprof(cfg.Pprof), OptSetTempDirPath(cfg.TempDirPath))
+	managerOpts := []pluginManagerOpt{
+		OptSetPprof(cfg.Pprof),
+		OptSetTempDirPath(cfg.TempDirPath),
+	}
+	if cfg.IsTLSEnabled() {
+		if cfg.RootCertPaths != "" {
+			certPaths := filepath.SplitList(cfg.RootCertPaths)
+			c.grpcSecurity = client.SecurityTLSExtended(cfg.TLSCertPath, cfg.TLSKeyPath, client.SecureClient, certPaths)
+		} else {
+			c.grpcSecurity = client.SecurityTLSEnabled(cfg.TLSCertPath, cfg.TLSKeyPath, client.SecureClient)
+		}
+	}
+	if cfg.IsTLSEnabled() {
```

Comment:
It could be moved into same conditional body.
fixed [here](https://github.com/intelsdi-x/snap/pull/1591/files/5e69931d5eb7c0449ac83cd4d00ea4d0a7c191d6..9365564f29304f562cd73499a34fe7178162402f#diff-0c3cd3b2fc4730963e170173bba333e6R234)